linker/shared/src/main/scala/org/scalajs/linker/standard/CommonPhaseConfig.scala

```diff
@@ -18,6 +18,8 @@ import org.scalajs.linker.interface._
 final class CommonPhaseConfig private (
     /** Core specification. */
     val coreSpec: CoreSpec,
+    /** Whether we are compiling to WebAssembly. */
+    val targetIsWebAssembly: Boolean,
```

Comment:
Hmmm... I see.

Form the looks of it, it more feels to me that the `coreSpec` is what the frontend and the backend need to agree on (at least this is what we check).

But maybe that notion is unnecessary anyways and we should plan to flatten it?

Reply:
If we take #5000 into account, then we could argue to `targetIsWebAssembly` becomes part of the core spec, since it changes what is linkable and what is not. :thinking:

Reply:
I now changed this to be in `CoreSpec`.